components/script/dom/globalscope.rs

```diff
@@ -1296,9 +1296,13 @@ impl GlobalScope {
 
     /// Returns the global scope for the given JSContext
     #[allow(unsafe_code)]
-    pub unsafe fn from_context(cx: *mut JSContext) -> DomRoot<Self> {
+    pub unsafe fn from_context(cx: *mut JSContext) -> Option<DomRoot<Self>> {
```

Comment:
Thinking about this further, I think we can do something more useful. `CurrentGlobalOrNull` uses the global from the current realm. This means that if we add an `InCompartment` argument to this method, we can force callers to provide proof that they have already entered a realm. Then we can remove the Option from the return value and assert that `global` is never null.

Reply:
Thanks! Good point and understood! Will update!